CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,187 @@
+1. Proposed - The exchange has been proposed, but not yet approved by the approver. The proposer may still cancel their proposal and be refunded their deposit.
+2. Approved - The exchange has been proposed and approved by the approver, but not yet executed or vetoed. The proposer may no longer cancel their proposal at this point.
+3. Executed - The exchange has been proposed, approved by the approver, and executed.
+4. Cancelled - The exchange has been cancelled by the proposer when it was in the Proposed state, or it has been vetoed by Governance when it was in the Approved stage.
+
+### Specific Changes:
+
+#### Modifications of existing contracts:
+
+##### `StableToken.sol`
+
+Currently, `StableToken.sol` only allows its `Exchange.sol` or `Validators.sol` to [mint](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L225) new stable tokens. Similarly, only its `Exchange.sol` can [burn](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L275) stable tokens.
+
+The following modifications will be made:
+1. **Modification to [mint()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L224)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted senders, Exchange and `Validators.sol`.
+2. **Modification to [burn()](https://github.com/celo-org/celo-monorepo/blob/master/packages/protocol/contracts/stability/StableToken.sol#L273)**
+   * Allow `msg.sender` to be `GrandaMento` in addition to the existing permitted sender, the Exchange.
+
+#### New contracts:
+
+##### `GrandaMento.sol`
+
+A new contract, `GrandaMento.sol`, is created, added to the Registry with the identifier `GrandaMento`, and owned by Governance. `GrandaMento` is a singleton contract that is capable of facilitating exchanges for multiple stable tokens-- that is, with the approval of Governance, `GrandaMento` could support `CELO <-> cUSD`, `CELO <-> cEUR`, and future `CELO <-> cXXX` exchanges. `GrandaMento` must be a spender of `Reserve.sol`, and supported stable tokens must allow `GrandaMento` to mint/burn them.
+
+The contract has the following configurable parameters:
+
+1. **`address public approver;`** - Set by Governance. Intended to be a multisig and has the authority to approve exchanges. The signers for the multisig are at the discretion of Governance, but it would comprise of community members or members on behalf of organizations that are aligned with the Celo network/ecosystem.
+2. **`uint256 public exchangeWaitPeriodSeconds;`** - Set by Governance. The minimum amount of time in seconds that must elapse between a proposed exchange being approved and when the exchange can be executed. Should give sufficient time for Governance to veto an approved exchange.
+3. **`FixidityLib.Fraction public spread;`** - Set by Governance. The percent fee imposed upon an exchange execution.
```

Comment:
I would lean towards giving as little control to the proposer as possible. More parameters add more complexity and could be a barrier to entry.

Some of the comparable solutions for large buyers also have very few input variables...
OTC trade: Similar intended conversion amount as input, variable conversion rate depending on OTC desk
Large limit order on CEX: variable limit price and quantity, uncertain (long or forever) execution time
Granda Mento: Should limit user controlled factors to just the total conversion amount. Conversion certainty in time is a benefit for the user over CEXes, conversion fee/rate certainty closer to proposal time is a benefit over OTC trades.

Reply:
Since the overall Granda Mento process will be gated by the community, are there any scenarios where two entities would want to propose a trade at a different spread after going through some sort of offline community discussion/pre-approval? e.g. an NGO vs a larger corporation looking to acquire cXXX.

Just thinking a bit here if we are sure we want to impose the same spread to all trades.

cc: @tkporter

Reply:
Hmm that's an interesting point, my feeling is that having the same spread is desirable from the community's perspective and that any large exchanger would be okay with having the same flat spread as that's sort of common in OTC. I think this is something that @albertclabs can probably attest to?